### PR TITLE
magento/magento2#: Replace deprecated Magento\Customer\Model\Customer::isConfirmationRequired method

### PR DESCRIPTION
**Pull request by atwixfirster**
### Description (*)

During new customer registration `Magento\Customer\Model\ResourceModel\Customer` model uses deprecated `Magento\Customer\Model\Customer`::`isConfirmationRequired` in the `_beforeSave()` method.

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

**Preconditions**

1. Set `Require Emails Confirmation` option to `Yes` on back-end side

back-end

**Steps to reproduce**
1. Create new customer

2. You will get a confirmation link in the email

**Expected result:** customer should get a confirmation email

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

Thank you!

### CODE REVIEW

File: app/code/Magento/Customer/Model/Customer.php [hunk 1]
```diff
@@ -805,7 +815,7 @@ public function isConfirmationRequired()
      */
     public function getRandomConfirmationKey()
     {
-        return md5(uniqid());
```
Comment (dmytro-ch) [inline app/code/Magento/Customer/Model/Customer.php:808]: Please, keep as it is.
Just ignore this rule:
```
// phpcs:ignore Magento2.Security.InsecureFunction
```

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
@atwixfirster, could you please also update the unit test for the session model?

File: app/code/Magento/Customer/Model/Customer.php (hunk 1 above)
Comment (atwixfirster) [reply app/code/Magento/Customer/Model/Customer.php:808]: Understood, @dmytro-ch!

**implemented**

Thank you!

**Comment by magento-engcom-team** [comment]
Hi @atwixfirster
Thank you for your contribution. This pull request has been combined (merged) with https://github.com/magento/magento2/pull/24340 to optimize pull requests queue on magento/magento2 repository
Closing this pull request.

**Pull request closed without merging**